SubmitUserActivites.cs

```diff
@@ -0,0 +1,342 @@
+            while (reader.Read())
+            {
+                int currID = reader.GetInt32("ID");
+                string currUserKey = reader.GetString("UserKey");
+
+                bool found = BCrypt.Net.BCrypt.Verify(userKey, currUserKey);
+                if (found) {
+                    ID = currID;
+                    break;
+                }
+            }
+            return ID;
+        }
+        public static bool SubmitActivities(int ID, List<dynamic> transports, List<dynamic> foods, List<dynamic> utilities) {
+            MySqlConnection connection = DatabaseConnecter.MySQLDatabase();
+            using(connection)
+            {
+                connection.Open();
+                MySqlTransaction transaction = connection.BeginTransaction();
+                try
+                {
+                    DateTime todaysDate = DateTime.UtcNow;
+                    string date = todaysDate.ToString("yyyy/MM/dd");
+
+                    // delete previous entries
+                    DeleteFromDatabase(connection, ID, date);
+
+                    // check if the new parameters are valid numbers
+                    bool validatedNumbers = ValidateNumbers(transports, foods, utilities);
+                    if (!validatedNumbers) {
+                        throw new Exception();
+                    }
+
+                    // input new entries using same query
+                    const string query = "INSERT INTO Activities (UserID, ActivityType, Method, Amount, DateTime, Emission) Values (@UserID, @ActivityType, @Method, @Amount, @DateTime, @Emission)";
+                    InputTransport(ID, transports, connection, date, query);
+                    InputFood(ID, foods, connection, date, query);
+                    InputUtilities(ID, utilities, connection, date, query);
+
+                    // update Daily emissions
+                    UpdateDailyEmissions(ID, connection, date);
+
+                    // commit and close
+                    transaction.Commit();
+                    connection.Close();
+                }
+                catch (Exception)
+                {
+                    transaction.Rollback();
+                    connection.Close();
+                    return false;
+                }
+            }
+            return true;
+        }
+        private static bool ValidateNumbers(List<dynamic> transports, List<dynamic> foods, List<dynamic> utilities)
+        {
+            // call each individual amount validator
+            bool transportsValid = ValidateTransportAmounts(transports);
+            bool foodsValid = ValidateFoodAmounts(foods);
+            bool utilitiesValid = ValidateUtilityAmounts(utilities);
+
+            // check if any are false, return false, otherwise true
+            return transportsValid && foodsValid && utilitiesValid;
+        }
+        private static bool ValidateTransportAmounts(List<dynamic> transports)

```

Comment:
Lets rethink a little bit of the request input object. If we are able to change the Amount/Miles/Hours into a single property name, we can easily put Validate___ into one function and differentiate with an enum or some string. This is the same for Input___ functions below.